database/skylink.go

```diff
@@ -21,22 +22,31 @@ var (
 	// ErrSkylinkNoExist is returned when we try to get a skylink that doesn't
 	// exist.
 	ErrSkylinkNoExist = errors.New("skylink does not exist")
+
+	// LockDuration defines the duration of a database lock. We lock skylinks
+	// while we are trying to pin them to a new server. The goal is to only
+	// allow a single server to pin a given skylink at a time.
+	LockDuration = 24 * time.Hour
```

Comment:
why not re-use the `mongo-lock` dependency we have in several places, collection already exists just have to pick a unique resource name

Reply:
these locks are currently not implemented yet right? couldn't find much references to it

Reply:
seems a bit high as well? is there a reasoning behind it being 24h?

Reply:
1. Using the locking library makes sense, I'll look into it.
2. Yes, the implementation is in #7. I split the full implementation into two PRs and some stuff has ended up in the first one while only being needed in the second.
3. 24 hours is fine because we want to scan these once per day at most. Once a skylink is locked by a certain server, it will be the first thing that server pulls from the DB until it's unlocked. So, if the server's execution terminates, it will pick it right back up.

Reply:
I'll actually add the locking library in #7 where the locks are actually used.